src/services/rsvp/rsvp-router.ts

```diff
@@ -0,0 +1,181 @@
+
+    //Returns error if query is empty
+    if (!queryResult) {
+        return res.status(StatusCode.ClientErrorBadRequest).send({ error: "UserNotFound" });
+    }
+
+    return res.status(StatusCode.SuccessOK).send(queryResult.toObject());
+});
+
+/**
+ * @api {get} /rsvp/ GET /rsvp/
+ * @apiGroup rsvp
+ * @apiDescription Check RSVP decision for current user, returns filtered info for attendees and unfiltered info for staff/admin
+ *
+ *
+ * @apiSuccess (200: Success) {string} MongoDB object ID
+ * @apiSuccess (200: Success) {string} userId
+ * @apiSuccess (200: Success) {string} User's applicatoin status
+ * @apiSuccess (200: Success) {string} User's Response (whether or whether not they're attending)
+ * @apiSuccessExample Example Success Response (caller is a user):
+ * 	HTTP/1.1 200 OK
+ *	{
+ *      "_id": "652c311b6e283244d2ef4c29",
+ *      "userId": "github0000001",
+ *      "status": "ACCEPTED",
+ *      "response": "ACCEPTED",
+ * 	}
+ *
+ *  @apiSuccessExample Example Success Response (caller is a staff/admin):
+ * 	HTTP/1.1 200 OK
+ *	{
+ *      "_id": "652c311b6e283244d2ef4c29",
+ *      "userId": "github0000001",
+ *      "status": "ACCEPTED",
+ *      "response": "ACCEPTED",
+ *      "reviewer": "reviewer1",
+ *      "emailSent": true,
+ * 	}
+ *
+ * @apiUse strongVerifyErrors
+ */
+rsvpRouter.get("/", strongJwtVerification, async (_: Request, res: Response) => {
+    const payload: JwtPayload = res.locals.payload as JwtPayload;
+
+    const userId: string = payload.id;
+
+    const queryResult: DecisionInfo | null = await Models.DecisionInfo.findOne({ userId: userId });
+
+    //Returns error if query is empty
+    if (!queryResult) {
+        return res.status(StatusCode.ClientErrorBadRequest).send({ error: "UserNotFound" });
+    }
+
+    const toReturn = queryResult.toObject();
+
+    //Filters data if caller doesn't have elevated perms
+    if (!hasElevatedPerms(payload)) {
+        delete toReturn.reviewer;
+        delete toReturn.emailSent;
+    }
+
+    return res.status(StatusCode.SuccessOK).send(toReturn);
+});
+
+/**
+ * @api {put} /rsvp/ Put /rsvp/
+ * @apiGroup rsvp
+ * @apiDescription Updates an rsvp for the currently authenticated user (determined by the JWT in the Authorization header).
+ *
+ * @apiBody {boolean} isAttending Whether or whether not the currently authenticated user is attending
+ * @apiParamExample {json} Example Request:
+ * {
+ *      "isAttending": false
+ * }
+ *
+ * @apiSuccess (200: Success) {string} MongoDB object ID
+ * @apiSuccess (200: Success) {string} userId
+ * @apiSuccess (200: Success) {string} User's applicatoin status
+ * @apiSuccess (200: Success) {string} User's Response (whether or whether not they're attending)
+ * @apiSuccess (200: Success) {string} Reviwer
+ * @apiSuccess (200: Success) {boolean} Whether email has been sent
+ * @apiSuccessExample Example Success Response:
+ * 	HTTP/1.1 200 OK
+ *	{
+ *      "_id": "652c311b6e283244d2ef4c29",
```

Comment:
_id shouldn't be mentioned in the docs, this is a confusing thing to include and long term we hope to remove it entirely